cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/BusinessMetadataDataset.java

```diff
@@ -85,12 +91,100 @@ public BusinessMetadataRecord getBusinessMetadata(Id.NamespacedId targetId, Stri
       return null;
     }
 
-    byte[] keyvalue = row.get(KEYVALUE_COLUMN);
     byte[] value = row.get(VALUE_COLUMN);
 
     return new BusinessMetadataRecord(targetId, key, Bytes.toString(value));
   }
 
+  /**
+   * Retrieves the business metadata for the specified {@link Id.NamespacedId}.
+   *
+   * @param targetId the specified {@link Id.NamespacedId}
+   * @return a Map representing the metadata for the specified {@link Id.NamespacedId}
+   */
+  public Map<String, String> getBusinessMetadata(Id.NamespacedId targetId) {
```

Comment:
In general, it's better to use some data structure/class rather than returning `Map<String, String>` because it'd be easier to define additional constraints and methods in future.